src/main/java/frc/trigon/robot/poseestimation/apriltagcamera/io/AprilTagLimelightIO.java

```diff
@@ -0,0 +1,70 @@
+package frc.trigon.robot.poseestimation.apriltagcamera.io;
+
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import frc.trigon.robot.constants.FieldConstants;
+import frc.trigon.robot.poseestimation.apriltagcamera.AprilTagCameraIO;
+import frc.trigon.robot.poseestimation.apriltagcamera.RobotPoseSourceInputsAutoLogged;
+import org.trigon.utilities.LimelightHelpers;
+
+public class AprilTagLimelightIO extends AprilTagCameraIO {
+    private final String hostname;
+
+    public AprilTagLimelightIO(String hostname) {
+        this.hostname = hostname;
+    }
+
+    @Override
+    protected void updateInputs(RobotPoseSourceInputsAutoLogged inputs) {
+        inputs.hasResult = LimelightHelpers.getTV(hostname);
+
+        if (inputs.hasResult)
+            updateHasResultInputs(inputs);
+        else
+            updateNoResultInputs(inputs);
+    }
+
+    private void updateHasResultInputs(RobotPoseSourceInputsAutoLogged inputs) {
+        final LimelightHelpers.Results results = LimelightHelpers.getLatestResults(hostname).targetingResults;
+        inputs.solvePNPPose = results.getBotPose3d_wpiBlue();
+        inputs.lastResultTimestamp = results.timestamp_LIMELIGHT_publish;
```

Comment:
You sure this will return in the same timeframe as the FPGA?

Reply:
I didn't change that part. It was your code.

Reply:
Might be wrong 
Double check
Fyi, I wrote this code two years ago, things have changed and I too make mistakes